packages/cst-dts-gen/api.d.ts

```diff
@@ -0,0 +1,12 @@
+import type { BaseParser } from "chevrotain"
```

Comment:
This import may need to be from the `@chevrotain/types` package to avoid cyclic deps.

Reply:
Cyclic deps may happen if we export the cst-dts-gen APIs from the `chevrotain` package.

Reply:
We won't be able to avoid a cyclic dependency as long as we use the non-serialized GAST.  `model.ts` imports the `GAstVisitor` from the `chevrotain` library.

Why would we want to export the cst-dts-gen API from the `chevrotain` package?  Couldn't users depend on it separately?  Most users probably wouldn't want it in their production code either.

Reply:
> Why would we want to export the cst-dts-gen API from the chevrotain package?

We won't want to do that long-term. But maybe in the short-term because the repo is in the middle of being split into a proper mono repo.

But you mentioned you can solve the API website issue with multiple entry points, so I hope we won't discover any other assumptions related to the "old" monolith approach.

Reply:
Should I put together a separate pull request?

Only thing I'm unsure of is if the github links will still work.  the typedoc doco suggested they wouldn't https://typedoc.org/guides/project-references/
i.e. where it mentions it's defined in the source file `Defined in api.d.ts:2473`